server/src/main/kotlin/org/javacs/kt/index/SymbolIndex.kt

```diff
@@ -103,29 +139,68 @@ class SymbolIndex {
         }
     }
 
-    private fun canStoreFqName(fqName: FqName) =
-           fqName.toString().length <= MAX_FQNAME_LENGTH
-        && fqName.shortName().toString().length <= MAX_SHORT_NAME_LENGTH
+    private fun Transaction.removeDeclarations(declarations: Sequence<DeclarationDescriptor>) =
+        declarations.forEach { declaration ->
+            val (descriptorFqn, extensionReceiverFqn) = getFqNames(declaration)
+
+            if (validFqName(descriptorFqn) && (extensionReceiverFqn?.let { validFqName(it) } != false)) {
+                Symbols.deleteWhere {
+                    (Symbols.fqName eq descriptorFqn.toString()) and (Symbols.extensionReceiverType eq extensionReceiverFqn?.toString())
+                }
+            } else {
+                LOG.warn("Excluding symbol {} from index since its name is too long", descriptorFqn.toString())
+            }
+        }
+
+    private fun Transaction.addDeclarations(declarations: Sequence<DeclarationDescriptor>) =
```

Comment:
Do we use anything from `Transaction`'s `this` in this block?

Reply:
Nice catch. Removed